### DISCUSSION TITLE
parse base64 string cost a lot of time

### DISCUSSION THREAD

**Discussion by 77h2l** [nlohmann/json#3202] [Q&A]
Hi,
when I convert a string into json type, using json::parse() api, my string contais a base64 encoded image,so the base64 string's length is very log ,nearly 1000000, and the parse api cost 200+ ms.
Is there any way to improve the performance？Thx

**Comment by nlohmann** [comment]
- Is this a release build?
- Is the JSON consisting of just one string? Could you share it to have a reference?

All the library is doing with strings is basically checking whether there are escape characters to work with (like `\uxxxx` or `\n`) and that the input is valid UTF-8. Apart of that, the input is copied to a `std::string`.

**Reply by 77h2l** [reply]
Amazing to get the reply so quickly. Thx a lot.
My json::parse() api input is like this:

[{"addition":{"barcode":{"barcode":"78209172869467","position":[{"x":"1509","y":"1797"},{"x":"1213","y":"1518"},{"x":"1179","y":"1555"},{"x":"1475","y":"1833"}]},"threecode":["341;7-13;002","341;7-13;004"...]},"extend":"","filename":"/home/linaro/78209172869467.jpg","image":{"colorspace":"RGB","encodeData":"/9j/2wBDAAUDBAQEAwUEBAQFBQUGBwwIBwcHBw8LCwkMEQ8SEhEPERETFhwXExQaFRERGCEYGh0dHx8fExciJCIeJBweHx7/2wBDAQUFBQcGBw4ICA4eFBEUHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh7/xAAfAAABBQEBAQEBAQAAAAAAAAAAAQIDBAUGBwgJCgv/xAC1EAACAQMDAgQDBQUEBAAAAX0BAgMABBEFEiExQQYTUWEHInEUMoGRoQgjQrHBFVLR8CQzYnKCCQoWFxgZGiUmJygpKjQ1Njc4OTpDREVGR0hJSlNUVVZXWFlaY2RlZmdoaWpzdHV2d3h5eoOEhYaHiImKkpOUlZaXmJmaoqOkpaanqKmqsrO0tba3uLm6wsPExcbHyMnK0tPU1dbX2Nna4eLj5OXm5+jp6vHy8/T19vf4+fr/xAAfAQADAQEBAQEBAQEBAAAAAAAAAQIDBAUGBwgJCgv/xAC1EQACAQIEBAMEBwUEBAABAncAAQIDEQQFITEGEkFRB2FxEyIygQgUQpGhscEJIzNS8BVictEKFiQ04SXxFxgZGiYnKCkqNTY3ODk6Q0RFRkdISUpTVFVWV1hZWmNkZWZnaGlqc3R1dnd4eXqCg4SFhoeIiYqSk5SVlpeYmZqio6Slpqeoqaqys7S1tre4ubrCw8TFxsfIycrS09TV1tfY2dri4+Tl5ufo6ery8/T19vf4+fr/wAARCAvcD7gDASIAAhEBAxEB/9oADAMBAAIRAxEAPwD49HWg9KDRj1oAB69KXj60YpO9AAelL3pDz9aKACl70DHNIfXHNAC4xSduKU84pB3FACjqPejNJ6Gl45oAWkFHUUDr60AHaj8aT1FL2oAPWjPFAo7UAL3NAAK+lNpV70AHeg9KD2p3G09c0ANPSgcUp6e9J1OKAEpW4IOaDRxgUAKOh5pD1FKO4pD9aAF74pB0FHf8KXsaADvSHuKO4NKw560AJ2BpV69KQA4oHWgAPalI5pD0pc9KADsRSdxSjuOhpD0oAD1NA6ClIAPNIBwaAAdaSl6EdqPUUALSd6UdKP4qADqKMYOBR60fWgBV5BpD1BoHUikPIoAX1oHQUHrQOlAB1ag9KO9LjrQAnpRjGaB0FH8VAB2BpR3pO1KOooAQfdo70oHUdM0dgaAE9RR6Up+8cUfwmgA6tikwcc0vejHWgA46igdxR2Bo/iNACenpTj1pp6daUdeaAAfdoHWgYAIoPbFAAR1oPOKG6mkHSgBV6mjsMcUDqKD3FACnqKQc5o9OvNB6mgAx0JpzfePpTe2aXvQAnb6Uv8VC9DSDqKADpkUvpQOpo/gH1oAQcEilwMZ70d+vakPQjFADj..."}}]

**Comment by 77h2l** [comment]
I have tested the api, even when the input jsut contains one key-value;
when the string length increases,the time cost increases.
Is there a high-performance way to parse this kind of input?

**Reply by nlohmann** [reply]
No, there is no alternative. If you could share the original data, I could profile it on my machine.
